src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```
@@ -0,0 +1,119 @@
+
+        duplicate.setLogin("vlad1996@gmail.com");
+
+        assertThrows(RegistrationException.class, () -> {
+            service.register(duplicate);
+        });
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        userExpected.setLogin(null);
+        storageDao.add(userExpected);
+
+        duplicate.setLogin(null);
+
+        assertThrows(RegistrationException.class, () -> {
+            service.register(duplicate);
+        });
+    }
+
+    @Test
+    void register_loginLessThenSix_notOk() {
+        userExpected.setLogin("vlad@gmail.com");
+        storageDao.add(userExpected);
+
+        duplicate.setLogin("vlad@gmail.com");
+
+        assertThrows(RegistrationException.class, () -> {
+            service.register(duplicate);
+        });
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        userExpected.setPassword(null);
+        storageDao.add(userExpected);
+
+        duplicate.setPassword(null);
+
+        assertThrows(RegistrationException.class, () -> {
+            service.register(duplicate);
+        });
+    }
+
+    @Test
+    void register_passwordLessThenSix_notOk() {
+        userExpected.setPassword("gggtt");
+        storageDao.add(userExpected);
+
+        duplicate.setPassword("gggtt");
+
+        assertThrows(RegistrationException.class, () -> {
+            service.register(duplicate);
+        });
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        userExpected.setAge(null);
+        storageDao.add(userExpected);
+
+        duplicate.setAge(null);
+
+        assertThrows(RegistrationException.class, () -> {
+            service.register(duplicate);
+        });
+    }
+
+    @Test
+    void register_ageLessThenEighteen_notOk() {
+        userExpected.setAge(17);
+        storageDao.add(userExpected);
+
+        duplicate.setAge(17);
+
+        assertThrows(RegistrationException.class, () -> {
+            service.register(duplicate);
+        });
+    }
+}
```

Add a test to ensure register(null) fails with the custom RegistrationException rather than causing a NullPointerException. The checklist warns: "Don't use `NullPointerException`, because in such case we can't distinguish was this exception because of our mistake or because of validation." Create `register_nullUser_notOk` which uses `assertThrows(RegistrationException.class, () -> service.register(null));`.